### ISSUE TITLE
Fields created in transformRecordHook can't be used by other config options

### ISSUE THREAD

**Issue by crccheck** [AntonBazhal/dynamo2es-lambda#48]
Because `transformRecordHook` comes at the end of processing, if you create a new field there, you can't use it for `typeField` or `indexField`, etc. I'm not sure if moving it up would create a chicken/egg problem.

https://github.com/AntonBazhal/dynamo2es-lambda/blob/88d43e98905b560aa42c75804bbb3e40d73e0b54/lib/handler.js#L76-L78

**Comment by iamdavidmartin** [comment]
Yeah, this would be super helpful. or this pull request: https://github.com/AntonBazhal/dynamo2es-lambda/pull/40.
